### PR TITLE
Add Yuncheng Lu as maintainer

### PR DESCRIPTION
**Pull request by VijayanB**
### Description
Add Yuncheng Lu as maintainer to k-NN repository
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VijayanB** [comment]
Tagging for review [@opensearch-project/plugins ](https://github.com/orgs/opensearch-project/teams/admin)

**Review by junqiu-lei — APPROVED** [review]

**Review by jmazanec15 — APPROVED** [review]

**Comment by VijayanB** [comment]
Will merge once i get approval from ticket

**Comment by VijayanB** [comment]
@luyuncheng Did you receive invitation to be outside collaborator?

**Comment by luyuncheng** [comment]
> @luyuncheng Did you receive invitation to be outside collaborator?

@VijayanB  yes, I have been accepted the invitation.

**Pull request merged**
